vulture_os/services/frontend/models.py

```diff
@@ -527,10 +533,90 @@ class Frontend(models.Model):
         verbose_name=_("Reclaim pending messages (ms)")
     )
     """ Performance settings """
+    queue_type = models.TextField(
+        default=RSYSLOG_QUEUE_TYPE_CHOICES[0][0],
+        choices=RSYSLOG_QUEUE_TYPE_CHOICES,
+        help_text=_("Set a queue type for the ruleset"),
+        verbose_name=_("Rsyslog queue type")
+    )
+    queue_size = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Size of the queue in nb of message"),
+        verbose_name=_("Size of the queue in nb of message"),
+        validators=[MinValueValidator(100)]
+    )
+    dequeue_batch_size = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Size of the batch to dequeue"),
+        verbose_name=_("Size of the batch to dequeue"),
+        validators=[MinValueValidator(1)]
+    )
     nb_workers = models.PositiveIntegerField(
         default=8,
         help_text=_("Maximum number of workers for rsyslog ruleset"),
-        verbose_name=_("Maximum parser workers")
+        verbose_name=_("Maximum parser workers"),
+        validators=[MinValueValidator(1)]
+    )
+    new_worker_minimum_messages = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Number of messages in queue to start a new worker thread"),
+        verbose_name=_("Minimum messages to start a new worker"),
+        validators=[MinValueValidator(1)]
+    )
+    shutdown_timeout = models.PositiveIntegerField(
+        default=5000,
+        blank=True,
+        null=True,
+        help_text=_("Time to wait for the queue to finish processing entries (in ms)"),
+        verbose_name=_("Queue timeout shutdown (ms)"),
+        validators=[MinValueValidator(1)]
+    )
+    enable_disk_assist = models.BooleanField(
+        default=False,
+        help_text=_("Enable disk-assisted queue on failure"),
+        verbose_name=_("Enable disk queue on failure")
+    )
+    high_watermark = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Write in DA queue after nb messages"),
+        verbose_name=_("High watermark target"),
+        validators=[MinValueValidator(100)]
+    )
+    low_watermark = models.PositiveIntegerField(
+        blank=True,
+        null=True,
+        help_text=_("Write in DA queue until nb messages"),
+        verbose_name=_("Low watermark target"),
+        validators=[MinValueValidator(100)]
+    )
```

Comment:
Maybe we should allow to configure watermarks using percentages of the size queue instead of hard values, for ease of use